src/dynamics/fixed_points/basic.lean

```diff
@@ -71,6 +71,24 @@ calc fb (g x) = g (fa x) : (h.eq x).symm
 protected lemma apply {x : α} (hx : is_fixed_pt f x) : is_fixed_pt f (f x) :=
 by convert hx
 
+@[simp] lemma _root_.set.preimage_iterate_eq {f : α → α} {n : ℕ} :
+  set.preimage (f^[n]) = ((set.preimage f)^[n]) :=
+begin
+  induction n with n ih, { simp, },
+  rw [iterate_succ, iterate_succ', set.preimage_comp', ih],
+end
+
+lemma is_fixed_pt_preimage_iff {s : set α} :
+  is_fixed_pt (set.preimage f) s ↔ f⁻¹' s = s :=
+iff.rfl
+
+lemma preimage_iterate_eq_self {s : set α} (h : f⁻¹' s = s) (n : ℕ) :
+  (f^[n])⁻¹' s = s :=
```

Comment:
Either move this lemma to `data.set.basic` or leverage the `is_fixed_pt` spelling.
```suggestion
lemma preimage_iterate {s : set α} (h : is_fixed_pt (set.preimage f) s) (n : ℕ) :
  is_fixed_pt (set.preimage (f^[n])) s :=
```
(also because you're in the `is_fixed_pt` namespace.

Reply:
OK done